waku/handshake.go

```diff
@@ -28,6 +35,35 @@ type statusOptions struct {
 	TopicInterest        []TopicType `rlp:"key=5"`
 }
 
+// initFLPKeyFields initialises the values of `idxFieldKey` and `keyFieldIdx`
+func initRLPKeyFields() {
+	o := statusOptions{}
+	v := reflect.ValueOf(o)
+
+	for i := 0; i < v.NumField(); i++ {
+		// skip unexported fields
+		if !v.Field(i).CanInterface() {
+			continue
+		}
+		rlpTag := v.Type().Field(i).Tag.Get("rlp")
+
+		// skip fields without rlp field tag
+		if rlpTag == "" {
+			continue
+		}
+
+		keys := strings.Split(rlpTag, "=")
+
+		if len(keys) != 2 || keys[0] != "key" {
+			panic("invalid value of \"rlp\" tag, expected \"key=N\" where N is uint")
+		}
```

Comment:
@cammellos Any objection to throwing a panic here, rather than throwing an error? I think a panic needs to be given, and in my mind it can be here or in the package `init()`.

Reply:
sure, panic looks fine